lnrpc/switchrpc/switch_server.go

```diff
@@ -573,6 +579,67 @@ func reconstructCircuit(sessionKey *btcec.PrivateKey,
 	}
 }
 
+// BuildOnion constructs a sphinx onion packet for the given route.
+func (s *Server) BuildOnion(_ context.Context,
+	req *BuildOnionRequest) (*BuildOnionResponse, error) {
+
+	if req.Route == nil {
+		return nil, status.Error(codes.InvalidArgument,
+			"route information is required")
+	}
+	if len(req.PaymentHash) == 0 {
+		return nil, status.Error(codes.InvalidArgument,
+			"payment hash is required")
+	}
+
+	var sessionKey *btcec.PrivateKey
+	var err error
+	if len(req.SessionKey) == 0 {
+		sessionKey, err = routing.GenerateNewSessionKey()
+		if err != nil {
+			return nil, status.Errorf(codes.Internal,
+				"failed to generate session key: %v", err)
+		}
+	} else {
+		if err := validateSessionKey(req.SessionKey); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument,
+				"invalid session key: %v", err)
+		}
+
+		sessionKey, _ = btcec.PrivKeyFromBytes(req.SessionKey)
+	}
+
+	// Convert the route to a Sphinx path.
+	route, err := s.cfg.RouteProcessor.UnmarshallRoute(req.Route)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"invalid route: %v", err)
+	}
+
+	// Generate the onion packet.
+	onionBlob, circuit, err := paymentsdb.GenerateSphinxPacket(
```

Comment:
I don't think we should leak the payments package into the switch rpc because of this function. Hoently we might think of moving this function somewhere else then the payments package, because it has zero dependencies to the payments packge and it was only a helper there. So I think its not good getting this dependency in, rather duplicate code than have this dependecy import.

Reply:
Yeah, it seems like a solid idea to move it. Checking back a ways, it used to be part of the `routing` package but was moved when addressing some stuck payment related issues.